modules/api/src/main/scala/com/github/scoquelin/arugula/commands/RedisSortedSetAsyncCommands.scala

```diff
@@ -672,11 +673,106 @@ object RedisSortedSetAsyncCommands {
   /**
    * A range of values
    *
-   * @param start The start value
-   * @param end The end value
+   * @param lower The lower bound
+   * @param upper The upper bound
    * @tparam T The value type
    */
-  final case class ZRange[T](start: T, end: T)
+  final case class ZRange[T](lower: ZRange.Boundary[T], upper: ZRange.Boundary[T])
+
+  object ZRange{
+    /**
+     * A boundary value, used for range queries (upper or lower bounds)
+     * @param value An optional value to use as the boundary. If None, it is unbounded.
+     * @param inclusive Whether the boundary is inclusive. default is false.
+     * @tparam T The value type
+     */
+    case class Boundary[T](value: Option[T] = None, inclusive: Boolean = false)
```

Comment:
After making Boundary type argument covariant [+T], this basically allows Boundary(Option[Nothing]) (which is basically the unbounded case) to be used together with other Boundary(Option[T]) and you can get rid of all the `new Boundary(...)` and new `ZRange(...)` in the code below